src/js/components/Drop/DropContainer.js

```diff
@@ -259,19 +259,25 @@ const DropContainer = forwardRef(
       const onClickDocument = (event) => {
         // determine which portal id the target is in, if any
         let clickedPortalId = null;
+        let clickedPortal = null;
         let node =
           containerTarget === document.body
             ? event.target
             : event?.composedPath()[0];
 
         while (clickedPortalId === null && node !== document) {
           const attr = node.getAttribute('data-g-portal-id');
-          if (attr !== null) clickedPortalId = parseInt(attr, 10);
+          if (attr !== null) {
+            clickedPortalId = parseInt(attr, 10);
+            clickedPortal = node;
+          }
           node = node.parentNode;
         }
+
         if (
           clickedPortalId === null ||
-          portalContext.indexOf(clickedPortalId) !== -1
+          portalContext.indexOf(clickedPortalId) !== -1 ||
+          (clickedPortal && clickedPortal !== (ref || dropRef).current)
```

Comment:
I think we might also need to check if `clickedPortal` is a child of `dropRef.current`. If you look at the Drop/Multiple story, there are examples of a drop within a drop. Clicking within the child drop closes the parent drop.